### PR TITLE
fix(types): update for selenium-webdriver types creating transpile errors

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

**Comment by cnishina** [comment]
There is a test that leaves a browser session open. Will investigate.

**Comment by cnishina** [comment]
Narrowed down to this test. It passes but the browser is still open: spec/driverProviderLocalConf.js.

We can fix this later.

**Comment by cnishina** [comment]
Also we are pinning the @types/selenium-webdriver version just to make sure we don't run into this problem again.

File: lib/driverProviders/mock.ts [hunk 1]
```diff
@@ -4,10 +4,15 @@
  * server.
  */
 import * as q from 'q';
+import {DeferredExecutor, Executor, Session, WebDriver} from 'selenium-webdriver';
```
Comment (juliemr) [inline lib/driverProviders/mock.ts]: DeferredExecutor, Executor, and Session look unused?

Comment (cnishina) [inline lib/driverProviders/mock.ts]: Nice catch. I had a few issues with mock last night... I'll remove this.

Comment (cnishina) [inline lib/driverProviders/mock.ts]: Removing `DeferredExecutor` and `Executor`. The Session constructor is used at `getNewDriver`.

**Comment by juliemr** [comment]
LGTM!

File: lib/expectedConditions.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {ProtractorBrowser} from './browser';
 import {ElementFinder} from './element';
 
-let webdriver = require('selenium-webdriver');
+const webdriver = require('selenium-webdriver');
```
Comment (sjelin) [inline lib/expectedConditions.ts]: nit: I'd prefer
```ts
import {error as wderror} from 'selenium-webdriver';
```

And changing line 166 accordingly

File: lib/locators.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import {By, promise as wdpromise, WebDriver, WebElement} from 'selenium-webdriver';
 
-let webdriver = require('selenium-webdriver');
+const webdriver = require('selenium-webdriver');
 let clientSideScripts = require('./clientsidescripts');
```
Comment (sjelin) [inline lib/locators.ts]: nit: in many places in this file, we use `webdriver.By`.  Prefer `By`.

File: lib/locators.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import {By, promise as wdpromise, WebDriver, WebElement} from 'selenium-webdriver';
 
-let webdriver = require('selenium-webdriver');
+const webdriver = require('selenium-webdriver');
```
Comment (sjelin) [inline lib/locators.ts]: nit: if you can address my other nit in this file, remove this declaration.

File: lib/ptor.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ export class Ptor {
   By: ProtractorBy;
   by: ProtractorBy;
   wrapDriver:
-      (webdriver: webdriver.WebDriver, baseUrl?: string, rootElement?: string,
+      (webdriver: WebDriver, baseUrl?: string, rootElement?: string,
        untrackOutstandingTimeouts?: boolean) => ProtractorBrowser;
   ExpectedConditions: ProtractorExpectedConditions;
```
Comment (sjelin) [inline lib/ptor.ts]: nit: below, for everything under `// Export selenium webdriver.`, prefer the imported version vs the `webdriver.` version

File: lib/ptor.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ export class Ptor {
   By: ProtractorBy;
```
Comment (sjelin) [inline lib/ptor.ts]: nit: up on line 11, prefer `const webdriver = ...`.  Or possibly just remove the declaration entirely if you can address my other nit in this file

File: lib/runner.ts [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import {EventEmitter} from 'events';
 import * as q from 'q';
+import {Session} from 'selenium-webdriver';
```
Comment (sjelin) [inline lib/runner.ts]: nit: add `promise as wdpromise` to this and replace all the instances of `webdriver.promise` below with `wdpromise`

File: lib/runner.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
...
 import * as util from 'util';
 
 import {ProtractorBrowser} from './browser';
```
Comment (sjelin) [inline lib/runner.ts]: nit: below on line 17, prefer `const webdriver = ...`.  Or, if you can address my other nit from this file, remove the declaration entirely.

File: package.json [hunk 9]
```diff
@@ -15,7 +15,7 @@
     "@types/jasmine": "^2.5.36",
     "@types/node": "^6.0.46",
     "@types/q": "^0.0.32",
-    "@types/selenium-webdriver": "~2.53.31",
+    "@types/selenium-webdriver": "2.53.31",
```
Comment (cnishina) [inline package.json]: Done.

File: lib/browser.ts [hunk 10]
```diff
@@ -562,7 +562,7 @@ export class ProtractorBrowser extends Webdriver {
    * @returns {!webdriver.promise.Promise} A promise that will be resolved to an
```
Comment (cnishina) [inline lib/browser.ts]: Scoping the require('selenium-webdriver') in only the for loop

File: lib/debugger.ts [hunk 11]
```diff
@@ -60,17 +61,15 @@ export class DebugHelper {
...
+        for (let i = 0; i < arr.length; ++i) {
+          arr[i].getText().then((text: string) => {
+            found.push(text);
           });
+        }
+        return found;
+      });
     };
     for (let key in global) {
       context[key] = global[key];
```
Comment (cnishina) [inline lib/debugger.ts]: done.

File: lib/debugger.ts [hunk 12]
```diff
@@ -230,9 +229,9 @@ export class DebugHelper {
    *     is done. The promise will resolve to a boolean which represents whether
```
Comment (cnishina) [inline lib/debugger.ts]: done.

File: lib/element.ts [hunk 13]
```diff
@@ -161,7 +161,7 @@ export class ElementArrayFinder extends WebdriverWebElement {
       if (this.getWebElements === null) {
```
Comment (cnishina) [inline lib/element.ts]: removed.

**Pull request merged**